<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>950</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>950</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>950</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="file_handling_section">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Image File</string>
    </property>
    <widget class="QPushButton" name="save_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>save</string>
     </property>
    </widget>
    <widget class="QPushButton" name="load_button">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>load</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="action_history_section">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>110</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Action History</string>
    </property>
    <widget class="QPushButton" name="undo_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Undo</string>
     </property>
    </widget>
    <widget class="QPushButton" name="redo_button">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Redo</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="zoom_section">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Zoom</string>
    </property>
    <widget class="QSlider" name="zoom_slider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>20</number>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="pageStep">
      <number>30</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="rotate_image_section">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>290</y>
      <width>181</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Rotate Image</string>
    </property>
    <widget class="QDial" name="dial">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>10</y>
       <width>50</width>
       <height>64</height>
      </rect>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="singleStep">
      <number>10</number>
     </property>
     <property name="pageStep">
      <number>20</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </widget>
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>30</y>
      <width>701</width>
      <height>531</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>689</width>
       <height>519</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="image_processing" name="image_canvas">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>30</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="image_processing" name="original_image">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>540</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="crop_image_section">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>380</y>
      <width>181</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Crop Image</string>
    </property>
    <widget class="QPushButton" name="apply_button">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>101</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Crop Canvas</string>
     </property>
    </widget>
   </widget>
   <widget class="image_processing" name="backup_image">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>540</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="reset_button">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>460</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>reset</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>image_processing</class>
   <extends>QLabel</extends>
   <header location="global">image_processing.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
